docker/pulumi/Dockerfile

```diff
@@ -1,12 +1,15 @@
 FROM debian:12 AS base
 
+ARG TARGETARCH
```

Comment:
This probably deserves a comment about which values are accepted here (`amd64` and `arm64` after this PR is merged if I'm not mistaken?)

Reply:
The TARGETARCH argument is automatically provided by docker. See: https://docs.docker.com/build/building/variables/#pre-defined-build-arguments
https://docs.docker.com/reference/dockerfile/#automatic-platform-args-in-the-global-scope

I could still make the requested updates if you think it is worth it.